### DISCUSSION TITLE
Wagmi cli: Allow using proxy implementation ABI on etherscan plugin

### DISCUSSION THREAD

**Discussion by brunowowk** [wevm/wagmi#1931] [Ideas]
When generating the sdk with contracts from etherscan plugin, if the contract is a proxy, there's no way to generate a sdk for the implementation instead of for the proxy. In those cases, we have to manually fetch the ABI and set it up without the Etherscan plugin.

There could be a field in the etherscan contract config to indicate if we want the proxy implementation

**Comment by tmm** [comment]
Good idea! Definitely would accept this contribution if you or anyone else wants to work on it.

**Reply by kadetXx** [reply]
Has this been implemented now?

**Reply by mrbasado** [reply]
For implementation of this what would you like to see? 

1) edit ContractConfig to include optional 'implementationAddress' field that if defined is used to fetch ABI.
	1) \+ very easy to implement
	2) \- requires user to dig for the information -> annoying + more errors

2) edit fetch to detect if the ABI is the proxy ABI then automatically call implementation() to get the real ABI. Would probably require a new field 'fetchImplementation' anyway that defaults to true but if set to false will return first ABI even if its the proxy one (otherwise getting that ABI will be annoying)
	1) \+ very automatic for user
	2) \- I'm just not sure all apiUrls support this method as I use mostly eth/sepolia

**Comment by foodaka** [comment]
any update here?

**Comment by songkeys** [comment]
For anyone interested, you can do it in your config file like this: https://github.com/RSS3-Network/Developer/blob/65babdd84b8951a1e6d8edd331c92dff9f325146/src/lib/wagmi/gen.ts#L11-L18
